src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -400,7 +401,7 @@ private void PrepareProjectExecutables()
             else
             {
                 exeSpec.ExecutionType = ExecutionType.Process;
-                if (Environment.GetEnvironmentVariable("DOTNET_WATCH") != "1")
+                if (environmentVariables.GetString("DOTNET_WATCH") != "1")
```

Comment:
Could this use GetBool? It automatically converts 1/0 to boolean.

Reply:
I considered that but it would be a breaking change as `GetBool` treats any non-zero value as true, whereas this treats any non-one value as false. I don't know how much of an issue this is, but it is a change.

Reply:
I searched GitHub for `DOTNET_WATCH` and only saw cases of it being set to `1`. I think it's fine to use the nicer extension method.